tls/s2n_handshake_io.c

```diff
@@ -694,7 +694,16 @@ static int s2n_advance_message(struct s2n_connection *conn)
 }
 
 int s2n_generate_new_client_session_id(struct s2n_connection *conn)
+    CONTRACT_REQUIRES(__CPROVER_is_fresh(conn, sizeof(*conn)) || (conn == NULL))
```

Comment:
Does this need to be part of the contract / included in our source code? What does it mean?

Reply:
At present it does need to be part of the contract, though I admit we could design something a little more compact. I am writing a proposal now for an improved feature for this.
What it means that we allow inputs that are either
1/ Null - because the code handles those
2/ Are a pointer to a chunk of memory of at least the specified size and which will be allocated by the contract checker.